code/obj/machinery/shield_generators/portable_shield_generator.dm

```diff
@@ -422,10 +440,12 @@
 	icon_state = "shieldw"
 	event_handler_flags = USE_FLUID_ENTER
 	var/powerlevel //Stores the power level of the deployer
+	var/isactive = TRUE
 	density = 0
 
 	var/sound/sound_shieldhit = "sound/impact_sounds/Energy_Hit_1.ogg"
 	var/obj/machinery/shieldgenerator/deployer = null
+	var/obj/machinery/door/linked_door = null
```

Comment:
Why is this on the forcefield parent instead of just the `doorlink` type?

Reply:
same for the associated stuff in `disposing`

Reply:
The current pathed doorlink-type forcefield is specifically one that's set up to doorlink *at map start* as a permanent attachment, and has code / subtype specific to that. The type still serves its role, but doorlinking is no longer a capability specific to that type.

Even if I were to implement a second energy-shield type specifically for use by shield generators for some reason, I would still want to have the doorlinking code included in the parent forcefield type so as to avoid duplicating the code.